### PR TITLE
Swift: Add MaD skeleton

### PR DESCRIPTION
**Pull request by MathiasVP**
This PR adds a skeleton for MaD for Swift. It's currently completely dead code, and a bunch of predicates are implemented as `none()` with a `// TODO` added next to it. Those predicates will need to be filled out before we can start to use the library. I prefer doing that in a different commit, though.

@rdmarsh2 I'd appreciate it if you could take a look at the changes in `DataFlowDispatch.qll` and `DataFlowPrivate.qll`. Other than that, I don't think a thorough review is required on this PR.

### CODE REVIEW

File: swift/ql/lib/codeql/swift/dataflow/internal/DataFlowDispatch.qll [hunk 1]
```diff
@@ -57,17 +61,78 @@ class DataFlowCallable extends TDataFlowCallable {
 
   /** Gets the location of this callable. */
   Location getLocation() { result = func.getLocation() }
+
+  AbstractFunctionDecl getUnderlyingCallable() { result = func }
```
Comment (rdmarsh2) [inline swift/ql/lib/codeql/swift/dataflow/internal/DataFlowDispatch.qll:65]: Should this return something compatible with closures?

Comment (MathiasVP) [reply swift/ql/lib/codeql/swift/dataflow/internal/DataFlowDispatch.qll:65]: Yeah, it probably should. How about we add `ClosureExpr` to https://github.com/github/codeql/blob/main/swift/ql/lib/codeql/swift/controlflow/internal/Scope.qll#L5 and use `Callable::TypeRange` instead?

Comment (MathiasVP) [reply swift/ql/lib/codeql/swift/dataflow/internal/DataFlowDispatch.qll:65]: https://github.com/github/codeql/pull/9557 makes the required changes for this to work. Once that PR is merged I'll fix this comment.

Comment (MathiasVP) [reply swift/ql/lib/codeql/swift/dataflow/internal/DataFlowDispatch.qll]: Fixed in eff046e2f75f6ab8ef48029ddbae97f95bad9504.

**Review by rdmarsh2 — APPROVED** [review]

**Pull request merged**